typescript/t.d.ts

```diff
@@ -294,4 +317,9 @@ export interface TFunction<Ns extends Namespace = DefaultNamespace, KPrefix = un
   ): TFunctionReturnOptionalDetails<TFunctionProcessReturnValue<$NoInfer<Ret>, DefaultValue>, TOpt>;
 }
 
+export type TFunction<
+  Ns extends Namespace = DefaultNamespace,
+  KPrefix = undefined,
+> = _StrictKeyChecks extends true ? TFunctionStrict<Ns, KPrefix> : TFunctionNonStrict<Ns, KPrefix>;
+
```

Comment:
Still not sure what are the possible implications of converting this `interface` into a `type` (although it will eventually resolve into an interface)